src/components/dialogs/generator-creation-dialog.js

```diff
@@ -379,7 +384,7 @@ const GeneratorCreationDialog = ({
                 currentNodeUuid,
                 generatorId,
                 sanitizeString(generatorName),
-                energySource,
+                !energySource ? UNDEFINED_ENERGY_SOURCE : energySource,
```

Comment:
May be eliminate the `!` by interverting then and else ?

Reply:
this code does not exist anymore